tracer/src/Datadog.Trace/Util/SamplingHelpers.cs

```diff
@@ -0,0 +1,15 @@
+// <copyright file="SamplingHelpers.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Util
+{
+    internal class SamplingHelpers
+    {
+        private const ulong KnuthFactor = 1_111_111_111_111_111_111;
+
+        internal static bool SampleByRate(ulong traceId, double rate) =>
```

Comment:
Yep according to the RFC you'll be passing in the span ID instead of the trace ID. Sounds good, renamed to `id` in 1ed8df4ef3f42c9bbd1f45a40de1510aecca1606